source/Posets/Delta-V-CompletePoset.lagda

```diff
@@ -0,0 +1,854 @@
+
+ module local-smallness (l : is-locally-small-order) where
+
+  _≤ⱽ_ : (x y : ∣ A ∣ₚ) → 𝓥  ̇
+  x ≤ⱽ y = pr₁ (l x y)
+
+  ≤ⱽ-is-prop : (x : ∣ A ∣ₚ) → (y : ∣ A ∣ₚ) → is-prop (x ≤ⱽ y)
+  ≤ⱽ-is-prop x y = equiv-to-prop (pr₂ (l x y)) (holds-is-prop (x ≤ y))
+
+  ≤ⱽ-≃-≤ : (x y : ∣ A ∣ₚ) → x ≤ⱽ y ≃ (x ≤ y) holds
+  ≤ⱽ-≃-≤ x y = pr₂ (l x y)
+
+  ≤ⱽ-to-≤ : (x y : ∣ A ∣ₚ) → x ≤ⱽ y → (x ≤ y) holds
+  ≤ⱽ-to-≤ x y =  ⌜ ≤ⱽ-≃-≤ x y ⌝
+
+  ≤-to-≤ⱽ : (x y : ∣ A ∣ₚ) → (x ≤ y) holds → x ≤ⱽ y
+  ≤-to-≤ⱽ x y = ⌜ ≤ⱽ-≃-≤ x y ⌝⁻¹
+
+module Retract-Lemmas (𝓤  𝓦  𝓥 : Universe) (A : Poset 𝓤 𝓦) where
+
+ open δ-complete-poset 𝓥 A
+ open PosetReasoning A
+ open non-trivial-posets A
+ open Positive-Posets 𝓤 𝓦 𝓥 A
+ open Local-Smallness 𝓤 𝓦 𝓥 A _≤_
+ open Joins (_≤_)
+
+ module def-Δ (i : is-δ-complete)
+              {x y : ∣ A ∣ₚ}
+              (x-≤-y : (x ≤ y) holds)
+               where
+
+  Δ : Ω 𝓥 → ∣ A ∣ₚ
+  Δ P = sup-of-δ i x y x-≤-y P
+
+ module retract-lemma₁ (l : is-locally-small-order)
+                       (i : is-δ-complete)
+                       (x y : ∣ A ∣ₚ)
+                       (x-≤-y : (x ≤ y) holds)
+                        where
+
+  open local-smallness l
+  open def-Δ i x-≤-y
+
+  non-trivial-to-Δ-section : x ≠ y → is-section (Δ ∘ Ω¬¬-to-Ω)
+  non-trivial-to-Δ-section x-≠-y = (r , H)
+   where
+    r : ∣ A ∣ₚ → Ω¬¬ 𝓥
+    r z = ((¬ (z ≤ⱽ x) , negations-are-props fe) , ¬-is-¬¬-stable)
+
+    f : ((P , P-¬¬-stable) : Ω¬¬ 𝓥) → ¬ (Δ P ≤ⱽ x) → P holds
+    f (P , P-¬¬-stable) not-Δ-≤-x = P-¬¬-stable not-not-P
+     where
+      not-not-P : ¬¬ (P holds)
+      not-not-P not-P = not-Δ-≤-x (≤-to-≤ⱽ (Δ P)
+                                  x
+                                  (transport (λ z → (z ≤ x) holds)
+                                             x-＝-Δ
+                                             (≤-is-reflexive A x)))
+       where
+        x-＝-Δ : x ＝ Δ P
+        x-＝-Δ = lower-＝-sup-δ i x y x-≤-y P not-P
+
+    g : ((P , P-¬¬-stable) : Ω¬¬ 𝓥) → P holds → ¬ (Δ P ≤ⱽ x)
+    g (P , P-¬¬-stable) in-P Δ-≤-x = x-≠-y (≤-is-antisymmetric A x-≤-y y-≤-x)
+     where
+      y-＝-Δ : y ＝ Δ P
+      y-＝-Δ = upper-＝-sup-δ i x y x-≤-y P in-P
+
+      y-≤-x : (y ≤ x) holds
+      y-≤-x = transport (λ z → (z ≤ x) holds)
+                        (y-＝-Δ ⁻¹)
+                        (≤ⱽ-to-≤ (Δ P) x Δ-≤-x)
+
+    H : r ∘ Δ ∘ Ω¬¬-to-Ω ∼ id
+    H (P , P-¬¬-stable) =
+      to-subtype-＝ (λ X → being-¬¬-stable-is-prop fe (holds-is-prop X))
+                    (to-subtype-＝ (λ Y → being-prop-is-prop fe)
+                                   (pe (negations-are-props fe)
+                                       (holds-is-prop P)
+                                       (f (P , P-¬¬-stable))
+                                       (g (P , P-¬¬-stable))))
+
+  Δ-section-to-non-trivial : is-section (Δ ∘ Ω¬¬-to-Ω) → x ≠ y
+  Δ-section-to-non-trivial (r , H) x-＝-y =
+    𝟘-is-not-𝟙 (ap (pr₁ ∘ pr₁) (r-x-＝-𝟘 ⁻¹ ∙ ap r x-＝-y ∙ r-y-＝-𝟙))
+   where
```

Comment:
It would be nice to use the equational chain reasoning syntax here, see e.g. https://www.cs.bham.ac.uk/~mhe/TypeTopology/UF.HiggsInvolutionTheorem.html#lc-monoid-structure-on-Ω-gives-EM